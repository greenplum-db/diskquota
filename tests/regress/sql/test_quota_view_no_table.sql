CREATE ROLE no_table SUPERUSER;

CREATE SCHEMA no_table;

SELECT diskquota.set_schema_quota('no_table', '1 MB');

SELECT schema_name, quota_in_mb, nspsize_in_bytes
FROM diskquota.show_fast_schema_quota_view;

SELECT diskquota.set_role_quota('no_table', '1 MB');

SELECT role_name, quota_in_mb, rolsize_in_bytes 
FROM diskquota.show_fast_role_quota_view;

SELECT diskquota.set_schema_tablespace_quota('no_table', 'pg_default', '1 MB');

SELECT schema_name, tablespace_name, quota_in_mb, nspsize_tablespace_in_bytes 
FROM diskquota.show_fast_schema_tablespace_quota_view;

SELECT diskquota.set_role_tablespace_quota('no_table', 'pg_default', '1 MB');

SELECT role_name, tablespace_name , quota_in_mb, rolsize_tablespace_in_bytes
FROM diskquota.show_fast_role_tablespace_quota_view;

DROP ROLE no_table;

DROP SCHEMA no_table;

-- Wait until the quota configs are removed from the memory 
-- automatically after DROP.
SELECT diskquota.wait_for_worker_new_epoch();
