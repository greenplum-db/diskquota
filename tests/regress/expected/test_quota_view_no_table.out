CREATE ROLE no_table SUPERUSER;
CREATE SCHEMA no_table;
SELECT diskquota.set_schema_quota('no_table', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

SELECT schema_name, quota_in_mb, nspsize_in_bytes
FROM diskquota.show_fast_schema_quota_view;
 schema_name | quota_in_mb | nspsize_in_bytes 
-------------+-------------+------------------
 no_table    |           1 |                0
(1 row)

SELECT diskquota.set_role_quota('no_table', '1 MB');
 set_role_quota 
----------------
 
(1 row)

SELECT role_name, quota_in_mb, rolsize_in_bytes 
FROM diskquota.show_fast_role_quota_view;
 role_name | quota_in_mb | rolsize_in_bytes 
-----------+-------------+------------------
 no_table  |           1 |                0
(1 row)

SELECT diskquota.set_schema_tablespace_quota('no_table', 'pg_default', '1 MB');
 set_schema_tablespace_quota 
-----------------------------
 
(1 row)

SELECT schema_name, tablespace_name, quota_in_mb, nspsize_tablespace_in_bytes 
FROM diskquota.show_fast_schema_tablespace_quota_view;
 schema_name | tablespace_name | quota_in_mb | nspsize_tablespace_in_bytes 
-------------+-----------------+-------------+-----------------------------
 no_table    | pg_default      |           1 |                           0
(1 row)

SELECT diskquota.set_role_tablespace_quota('no_table', 'pg_default', '1 MB');
 set_role_tablespace_quota 
---------------------------
 
(1 row)

SELECT role_name, tablespace_name , quota_in_mb, rolsize_tablespace_in_bytes
FROM diskquota.show_fast_role_tablespace_quota_view;
 role_name | tablespace_name | quota_in_mb | rolsize_tablespace_in_bytes 
-----------+-----------------+-------------+-----------------------------
 no_table  | pg_default      |           1 |                           0
(1 row)

DROP ROLE no_table;
DROP SCHEMA no_table;
-- Wait until the quota configs are removed from the memory 
-- automatically after DROP.
SELECT diskquota.wait_for_worker_new_epoch();
 wait_for_worker_new_epoch 
---------------------------
 t
(1 row)

